### PR TITLE
adds ruby with devkit 3.2

### PR DESCRIPTION
**Pull request by baywet**
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/AUTHORING_MANIFESTS.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.2 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.2.0)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

-----

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-pkgs/pull/92456)

### CODE REVIEW

**Comment by baywet** [comment]
This is caused by changes in the installed, I've created an issue to get to the bottom of that https://github.com/oneclick/rubyinstaller2/issues/326

**Review by mdanish-kh — CHANGES_REQUESTED** [review]
Must match Registry values

File: manifests/r/RubyInstallerTeam/RubyWithDevKit/3/2/3.2.0-1/RubyInstallerTeam.RubyWithDevKit.3.2.installer.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+PackageVersion: 3.2.0-1
+MinimumOSVersion: 10.0.0.0
+Scope: machine
+InstallModes:
+- interactive
+- silent
+- silentWithProgress
+InstallerSwitches:
+  Silent: /verysilent /allusers
+  SilentWithProgress: /silent /allusers
```
Comment (mdanish-kh) [inline manifests/r/RubyInstallerTeam/RubyWithDevKit/3/2/3.2.0-1/RubyInstallerTeam.RubyWithDevKit.3.2.installer.yaml:14]:
```suggestion
```
We don't need silent switches when Installer type is set to `inno`. We'll add scope specific flags at the installer level.

File: manifests/r/RubyInstallerTeam/RubyWithDevKit/3/2/3.2.0-1/RubyInstallerTeam.RubyWithDevKit.3.2.installer.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
+# Created using wingetcreate 1.1.2.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.installer.1.2.0.schema.json
+
+PackageIdentifier: RubyInstallerTeam.RubyWithDevKit.3.2
+PackageVersion: 3.2.0-1
+MinimumOSVersion: 10.0.0.0
+Scope: machine
```
Comment (mdanish-kh) [inline manifests/r/RubyInstallerTeam/RubyWithDevKit/3/2/3.2.0-1/RubyInstallerTeam.RubyWithDevKit.3.2.installer.yaml:7]:
```suggestion
```
Going to add scope at the installer level and pass the switches there.

File: manifests/r/RubyInstallerTeam/RubyWithDevKit/3/2/3.2.0-1/RubyInstallerTeam.RubyWithDevKit.3.2.installer.yaml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+UpgradeBehavior: install
+FileExtensions:
+- rb
+- rbw
+ReleaseDate: 2022-04-26
+Installers:
+- Architecture: x64
+  InstallerType: inno
+  InstallerUrl: https://github.com/oneclick/rubyinstaller2/releases/download/rubyinstaller-3.2.0-1/rubyinstaller-devkit-3.2.0-1-x64.exe
+  InstallerSha256: D82313EED91F3210C3CFAC42214497477F6919A540C6CA0F237D25C3810040E2
```
Comment (mdanish-kh) [inline manifests/r/RubyInstallerTeam/RubyWithDevKit/3/2/3.2.0-1/RubyInstallerTeam.RubyWithDevKit.3.2.installer.yaml:24]:
```suggestion
  InstallerSha256: D82313EED91F3210C3CFAC42214497477F6919A540C6CA0F237D25C3810040E2
  Scope: machine
  InstallerSwitches:
    Custom: /ALLUSERS
```

File: manifests/r/RubyInstallerTeam/RubyWithDevKit/3/2/3.2.0-1/RubyInstallerTeam.RubyWithDevKit.3.2.installer.yaml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  AppsAndFeaturesEntries:
+  - DisplayName: Ruby 3.2.0-1-x64-ucrt with MSYS2
+    ProductCode: RubyInstaller-3.2-x64-mingw-ucrt_is1
```
Comment (mdanish-kh) [inline manifests/r/RubyInstallerTeam/RubyWithDevKit/3/2/3.2.0-1/RubyInstallerTeam.RubyWithDevKit.3.2.installer.yaml:27]:
```suggestion
    ProductCode: RubyInstaller-3.2-x64-mingw-ucrt_is1
- Architecture: x64
  InstallerType: inno
  InstallerUrl: https://github.com/oneclick/rubyinstaller2/releases/download/rubyinstaller-3.2.0-1/rubyinstaller-devkit-3.2.0-1-x64.exe
  InstallerSha256: D82313EED91F3210C3CFAC42214497477F6919A540C6CA0F237D25C3810040E2
  AppsAndFeaturesEntries:
  - DisplayName: Ruby 3.2.0-1-x64
    ProductCode: RubyInstaller-3.2-x64-mingw-ucrt_is1
  Scope: user
  InstallerSwitches:
    Custom: /ALLUSERS
```
Another installer entry for user scope.

File: manifests/r/RubyInstallerTeam/RubyWithDevKit/3/2/3.2.0-1/RubyInstallerTeam.RubyWithDevKit.3.2.installer.yaml [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+- Architecture: x86
+  InstallerType: inno
+  InstallerUrl: https://github.com/oneclick/rubyinstaller2/releases/download/rubyinstaller-3.2.0-1/rubyinstaller-devkit-3.2.0-1-x86.exe
+  InstallerSha256: 891C0F05CDA258049C36ACEF1401B5DD19A7EFB277108D2170DB34C6770AE282
```
Comment (mdanish-kh) [inline manifests/r/RubyInstallerTeam/RubyWithDevKit/3/2/3.2.0-1/RubyInstallerTeam.RubyWithDevKit.3.2.installer.yaml:31]:
```suggestion
  InstallerSha256: 891C0F05CDA258049C36ACEF1401B5DD19A7EFB277108D2170DB34C6770AE282
  Scope: machine
  InstallerSwitches:
    Custom: /ALLUSERS
```

File: manifests/r/RubyInstallerTeam/RubyWithDevKit/3/2/3.2.0-1/RubyInstallerTeam.RubyWithDevKit.3.2.installer.yaml [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+  AppsAndFeaturesEntries:
+  - DisplayName: Ruby 3.2.0-1-x86-msvcrt with MSYS2
+    ProductCode: RubyInstaller-3.2-i386-mingw32_is1
```
Comment (mdanish-kh) [inline manifests/r/RubyInstallerTeam/RubyWithDevKit/3/2/3.2.0-1/RubyInstallerTeam.RubyWithDevKit.3.2.installer.yaml:34]:
```suggestion
    ProductCode: RubyInstaller-3.2-i386-mingw32_is1
- Architecture: x86
  InstallerType: inno
  InstallerUrl: https://github.com/oneclick/rubyinstaller2/releases/download/rubyinstaller-3.2.0-1/rubyinstaller-devkit-3.2.0-1-x86.exe
  InstallerSha256: 891C0F05CDA258049C36ACEF1401B5DD19A7EFB277108D2170DB34C6770AE282
  AppsAndFeaturesEntries:
  - DisplayName: Ruby 3.2.0-1-x86
    ProductCode: RubyInstaller-3.2-i386-mingw32_is1
  Scope: user
  InstallerSwitches:
    Custom: /CURRENTUSER
```
Another installer entry for user scope.

**Review by mdanish-kh — CHANGES_REQUESTED** [review]

**Comment by baywet** [comment]
@mdanish-kh thanks for the suggestions!

**Comment by mdanish-kh** [comment]
@baywet Looks like this version just got pulled by the publisher :/ https://github.com/oneclick/rubyinstaller2/releases, The URLs were working an hour or so earlier.

**Comment by baywet** [comment]
@mdanish-kh noticed that as well. Let's give them a day or two to put it back up, I'll update the hashes if necessary.

**Comment by larskanis** [comment]
Sorry, I published the binaries, before they were final. But now they are.

**Comment by baywet** [comment]
@larskanis no worries, thanks for letting us know. I just pushed an update with the updated URLs + hashes. Hopefully this one is the right one, thanks everyone for the help!

**Review by mdanish-kh — CHANGES_REQUESTED** [review]
@baywet I'm wondering whether this package should now be under `manifests/r/RubyInstallerTeam/Ruby/3/2` as I don't see `devkit` in the InstallerUrl 👀

**Comment by baywet** [comment]
@mdanish-kh good catch, bad copy paste on my side, fixed.

**Review by mdanish-kh — APPROVED** [review]
No worries :) Looks like we're good to go now

**Pull request merged**